impala/tests/test_data_types.py

```diff
@@ -120,3 +131,24 @@ def test_utf8_strings(cur):
     result = cur.fetchone()[0]
     assert result == b"\xaa"
     assert result.decode("UTF-8", "replace") == u"�"
+
+@pytest.mark.connect
+def test_utf8_strings_no_string_conv(cur_no_string_conv):
+    cur = cur_no_string_conv
+    """Use STRING/VARCHAR/CHAR values  with multi byte unicode code points in a query."""
```

Comment:
I don't think that this complex unicode handling is needed for the "no conversion" case. The main point should be to check that with unicode conversion the type returned is unicode, while without it it is bytes (in Python 3)

Reply:
Got it